app/src/main/java/de/xikolo/utils/extensions/IntentExtensions.kt

```diff
@@ -46,19 +53,90 @@ fun <T : Intent> T.createChooser(context: Context, title: String? = null, packag
     }
 }
 
+private fun <T : Intent> T.open(
+    activity: FragmentActivity,
+    forceChooser: Boolean,
+    parentFile: File? = null
+): Boolean {
+    val chooserIntent = createChooser(App.instance)
+
+    fun startActivity(): Boolean {
+        return try {
+            activity.startActivity(
+                if (forceChooser) {
+                    chooserIntent
+                } else {
+                    this
+                }
+                    ?.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
```

Comment:
Maybe add `this` before the `?` to make it more clear for the reader.

Reply:
The statement belongs to the if-else block above

Reply:
Ahh, looks more intuitive now, thanks!